ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -555,6 +562,44 @@ func (ipfs *Connector) RepoStat() (api.IPFSRepoStat, error) {
 	return stats, nil
 }
 
+// Resolve accepts ipfs or ipns path and resolves it into a cid
+func (ipfs *Connector) Resolve(path string) (cid.Cid, error) {
+	validPath, err := gopath.ParsePath(path)
+	if err != nil {
+		logger.Error("could not parse path: " + err.Error())
+		return cid.Undef, err
+	}
+
+	if !strings.HasPrefix(path, "/ipns") && validPath.IsJustAKey() {
+		ci, _, err := gopath.SplitAbsPath(validPath)
+		return ci, err
+	}
+
+	ctx, cancel := context.WithTimeout(ipfs.ctx, ipfs.config.IPFSRequestTimeout)
+	defer cancel()
+	res, err := ipfs.postCtx(ctx, "resolve?arg="+url.QueryEscape(path), "", nil)
+	if err != nil {
+		logger.Error(err)
+		return cid.Undef, err
+	}
+
+	var resp ipfsResolveResp
+	err = json.Unmarshal(res, &resp)
+	if err != nil {
+		logger.Error("could not unmarshal response: " + err.Error())
+		return cid.Undef, err
+	}
+
+	validPath, err = gopath.ParsePath(resp.Path)
```

Comment:
You can skip this. Not only I'd expect a valid path coming in the ipfs response, but if `SplitAbsPath` gives you a valid CID you should be ok.